tools/tests/test_check_rabbitmq_queue.py

```diff
@@ -6,16 +6,42 @@
 
 class AnalyzeQueueStatsTests(TestCase):
     def test_no_stats_available(self) -> None:
-        result = analyze_queue_stats("name", {}, 0)
+        result = analyze_queue_stats("name", {}, {}, 0)
         self.assertEqual(result["status"], UNKNOWN)
 
     def test_queue_stuck(self) -> None:
         """Last update > 5 minutes ago and there's events in the queue."""
 
-        result = analyze_queue_stats("name", {"update_time": time.time() - 301}, 100)
+        result = analyze_queue_stats("name", {"update_time": time.time() - 301}, {}, 100)
         self.assertEqual(result["status"], CRITICAL)
         self.assertIn("queue appears to be stuck", result["message"])
 
+    def test_queue_stuck_in_restart_loop(self) -> None:
+        """
+        Worker keeps restarting and crashing without being able to process events,
+        meaning the .old stats file gets relatively old while current stats file stays
+        with a fresh timestamp.
+        """
+
+        now = time.time()
+        result = analyze_queue_stats(
+            "name",
+            {
+                "update_time": now,
+                "current_queue_size": 10000,
+                "recent_average_consume_time": None,
+            },
+            {
+                "update_time": now - 200,
+                "current_queue_size": 10000,
+                "recent_average_consume_time": None,
```

Comment:
Should this value be nonzero here?

Reply:
Yeah I think so - this value is currently irrelevant and not looked at, but indeed it doesn't make sense to have `None` here - fixed.